plugins/packages/polygon/main.go

```diff
@@ -100,3 +99,47 @@ func getBlockAuthor(header *evm.Header) (common.Address, error) {
 
 	return signer, nil
 }
+
+var (
+	polygonHitMeter  = metrics.NewMajorMeter("/flume/polygon/hit")
+	polygonMissMeter = metrics.NewMajorMeter("/flume/polygon/miss")
+)
+
+func borBlockDataPresent(input interface{}, cfg *config.Config, db *sql.DB) bool {
+	present := true
+	switch input.(type) {
+	case plugins.BlockNumber:
+		if uint64(input.(plugins.BlockNumber).Int64()) < cfg.EarliestBlock {
+			present = false
+			return present
+		}
+	case types.Hash:
+		blockHash := input.(types.Hash)
+		var response int
+		statement := "SELECT 1 FROM bor.bor_logs WHERE blockHash = ?;"
+		db.QueryRow(statement, plugins.TrimPrefix(blockHash.Bytes())).Scan(&response)
+		if response == 0 {
+			present = false
+			return present
+		}
+	}
+	return present
+}
+
+func borTxDataPresent(txHash types.Hash, cfg *config.Config, db *sql.DB) bool {
+	present := true
+	var response int
+	txStatement := "SELECT 1 FROM bor.bor_logs WHERE transactionHash = ?;"
+	db.QueryRow(txStatement, plugins.TrimPrefix(txHash.Bytes())).Scan(&response)
+	if response == 0 {
+		present = false
+		return present
+	}
+	mpStatement := "SELECT 1 FROM mempool.transactions WHERE hash = ?;"
```

Comment:
Bor transactions will never be on the mempool table, so I don't think we need to check here for bor specific things.